File: Array_and_Hashing/CPP/reverse_string.cpp
#include<bits/stdc++.h>
using namespace std;


int main (){


    string s = "]ecitcarP smhtiroglA arodef-derahs@rekcah[";
    int i = 0 , j = size(s) - 1;
    char temp;
    while(i <= j) temp = s[i], s[i] = s[j], s[j] = temp ,i++, j--;   
    cout<<s<<endl;
}

File: Array_and_Hashing/JS/ReverseString.js
// Not working 

init();

function swap(r, x, y) {
    let temp;
    temp = r[x];
    r[x] = r[y];
    r[y] = temp;
    return r;
}

function reverse(s) {

    let len = s.length;
    let r = s.slice();
    let temp = '',
        i = 0,
        j = len - 1;

    while (i <= j) swap(r, i, j), i++, j--;

    console.log(r);

    return r;
}



function init() {

    let s = "]ecitcarP smhtiroglA arodef-derahs@rekcah[";

    let r = reverse(s);

    console.log(r);
}